### PR TITLE
delete tag

### PR DESCRIPTION
**Pull request by critical27**
common/storage code has been merged before.

Supported syntax:
```
# single tag
DELETE TAG t1 FROM "a";
# multiple tag
DELETE TAG t1,t2 FROM"a" ;
# all tag
DELETE TAG * FROM "a";
# delete multiple vertices's tag 
DELETE TAG t1 FROM "a", "b";
```
Pipe and var is supported as well.

### CODE REVIEW

File: tests/tck/features/delete/DeleteTag.feature [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+      """
+    Then the result should be, in any order:
+      | VertexID     | bachelor.name | bachelor.speciality |
+      | "Tim Duncan" | "Tim Duncan"  | "psychology"        |
+    When executing query:
+      """
+      LOOKUP ON player WHERE player.name == "Tim Duncan"
+      """
+    Then the result should be, in any order:
+      | VertexID |
```
Comment (yixinglu) [inline tests/tck/features/delete/DeleteTag.feature:57]: Then drop the used space at the end of the scenario.

File: tests/tck/features/delete/DeleteTag.feature [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+      FETCH PROP ON bachelor "Tim Duncan" YIELD bachelor.name, bachelor.speciality
+      """
+    Then the result should be, in any order:
+      | VertexID | bachelor.name | bachelor.speciality |
+    When executing query:
+      """
+      LOOKUP ON player WHERE player.name == "Tim Duncan"
+      """
+    Then the result should be, in any order:
+      | VertexID |
```
Comment (yixinglu) [inline tests/tck/features/delete/DeleteTag.feature:108]: ditto in following scenarios.

File: tests/tck/features/delete/DeleteTag.feature [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+      """
+      FETCH PROP ON team "Spurs" YIELD team.name
+      """
+    Then the result should be, in any order:
+      | VertexID | team.name |
+      | "Spurs"  | "Spurs"   |
+    # delete one tag
+    When executing query:
+      """
+      GO FROM "Tim Duncan" OVER serve YIELD serve._dst as id | DELETE TAG team FROM $-.id
```
Comment (yixinglu) [inline tests/tck/features/delete/DeleteTag.feature:246]: Test following cases:

```
GO FROM "Tim Duncan" OVER serve YIELD serve._dst as id | DELETE TAG team FROM $-.id, "Lakers"
$var = GO FROM "Tim Duncan" OVER serve YIELD serve._dst as id; DELETE TAG team FROM $var.id, "Lakers"
```

File: src/graph/validator/MutateValidator.h [hunk 4]
```diff
@@ -85,6 +85,28 @@ class DeleteVerticesValidator final : public Validator {
...
+    Status toPlan() override;
+
+private:
+    GraphSpaceID                                  spaceId_{-1};
+    // From ConstantExpression
+    std::vector<Value>                            vertices_;
+    // From InputPropertyExpression or InputPropertyExpression
+    Expression*                                   vidRef_{nullptr};
+    std::vector<TagID>                            tagIds_;
+};
```
Comment (yixinglu) [inline src/graph/validator/MutateValidator.h:108]: format

File: tests/tck/features/delete/DeleteTag.feature (hunk 3 above)
Comment (yixinglu) [reply tests/tck/features/delete/DeleteTag.feature:246]: If not support above cases, please report error and add these scenarios in TCK

Comment (critical27) [reply tests/tck/features/delete/DeleteTag.feature:250]: 🆗 , done

File: src/graph/validator/MutateValidator.h (hunk 4 above)
Comment (critical27) [reply src/graph/validator/MutateValidator.h:106]: done

File: tests/tck/features/delete/DeleteTag.feature (hunk 1 above)
Comment (critical27) [reply tests/tck/features/delete/DeleteTag.feature:57]: done

**Review by yixinglu — APPROVED** [review]

**Review by bright-starry-sky — APPROVED** [review]

**Pull request merged**